massa-factory-worker/src/tests/block_scenarios.rs

```diff
@@ -0,0 +1,184 @@
+use std::{str::FromStr, sync::Arc};
+
+use super::BlockTestFactory;
+use massa_consensus_exports::MockConsensusController;
+use massa_hash::Hash;
+use massa_models::{
+    address::Address,
+    amount::Amount,
+    block_id::BlockId,
+    config::THREAD_COUNT,
+    operation::{Operation, OperationSerializer, OperationType},
+    secure_share::SecureShareContent,
+    slot::Slot,
+};
+use massa_pool_exports::MockPoolController;
+use massa_pos_exports::MockSelectorController;
+use massa_signature::KeyPair;
+use massa_storage::Storage;
+use parking_lot::{Condvar, Mutex};
+use serial_test::serial;
+
+/// Creates a basic empty block with the factory.
+#[test]
+#[serial]
+fn basic_creation() {
+    let default_panic = std::panic::take_hook();
+    std::panic::set_hook(Box::new(move |info| {
+        default_panic(info);
+        std::process::exit(1);
+    }));
+    let keypair = KeyPair::generate(0).unwrap();
+    let storage = Storage::create_root();
+    let staking_address = Address::from_public_key(&keypair.get_public_key());
+    let parent = BlockId::generate_from_hash(Hash::compute_from("test".as_bytes()));
+    let mut parents = Vec::new();
+    for i in 0..THREAD_COUNT as u64 {
+        parents.push((parent, i));
+    }
+    let pair = Arc::new((Mutex::new(false), Condvar::new()));
+    let pair2 = pair.clone();
+    let mut consensus_controller = Box::new(MockConsensusController::new());
+    consensus_controller
+        .expect_get_best_parents()
+        .times(1)
+        .return_once(move || parents);
+    consensus_controller
+        .expect_register_block()
+        .times(1)
+        .return_once(move |_, _, storage, created| {
+            assert!(created);
+            let block = storage.get_block_refs();
+            assert_eq!(block.len(), 1);
+            let &(ref lock, ref cvar) = &*pair2;
```

Comment:
This is weird I'm pretty sure you don't need to do that as `let (ref lock, ref cvar) = *pair2;` would work the same. You've done it multiple time but LGTM besides that 👍

Reply:
Done I have updated following the example in the std : https://doc.rust-lang.org/stable/std/sync/struct.Condvar.html#examples-2 dunno why i had these refs.